CMakeLists.txt

```diff
@@ -9,6 +9,9 @@ include(CMakeDependentOption)
 
 option(USE_MPI "Build with MPI support" ON)
 option(USE_OPENMP "Build with OpenMP support" ON)
+cmake_dependent_option(USE_OFFLOAD "Offloading to devices if OpenMP is enabled" OFF "USE_OPENMP" OFF)
+set(WITH_OFFLOAD_FLAGS "" CACHE STRING "Additional flag(s) if OpenMP/offload is enabled")
+set(WITH_OPENMP_FLAGS "" CACHE STRING "To replace OpenMP default flag, e.g., to replace -fopenmp")
```

Comment:
how does this differ from the possibility to set `OpenMP_Fortran_FLAG`, `OpenMP_C_FLAG`, etc. except that it sets the flag for all languages? See https://github.com/Kitware/CMake/blob/5f630a934aa4b8616aa59988ea70c14065d3a08d/Modules/FindOpenMP.cmake#L94

Reply:
Thank you, I will lookup this reference!  I expect the CMake part to cleanup along with our evolving compiler, and more/alternative toolchains. So far, the CMake portion of this PR is small and may get even smaller. Even with some of the (obviously) temporary additions, I expect no interference with the way things are working and being tested.